lib/src/process_helper/linux.rs

```diff
@@ -1,267 +1,11 @@
-    let mut command = Command::new("sh");
-    command.arg("-c").arg(command_string);
-
-    command
-}
-
-fn map_action_to_signal(action: &ProcessAction) -> Signal {
-    match action {
-        ProcessAction::Pause => Signal::SIGSTOP,
-        ProcessAction::Resume => Signal::SIGCONT,
-    }
-}
-
-fn map_internal_signal_to_nix_signal(signal: InternalSignal) -> Signal {
-    match signal {
-        InternalSignal::SigKill => Signal::SIGKILL,
-        InternalSignal::SigInt => Signal::SIGINT,
-        InternalSignal::SigTerm => Signal::SIGTERM,
-        InternalSignal::SigCont => Signal::SIGCONT,
-        InternalSignal::SigStop => Signal::SIGSTOP,
-    }
-}
-
-/// Convenience wrapper around `send_signal_to_child` for raw unix signals.
-/// Its purpose is to hide platform specific logic.
-pub fn send_internal_signal_to_child(
-    child: &Child,
-    signal: InternalSignal,
-    send_to_children: bool,
-) -> Result<bool> {
-    let signal = map_internal_signal_to_nix_signal(signal);
-    send_signal_to_child(child, signal, send_to_children)
-}
-
-/// Convenience wrapper around `send_signal_to_child` for internal actions on processes.
-/// Its purpose is to hide platform specific logic.
-pub fn run_action_on_child(
-    child: &Child,
-    action: &ProcessAction,
-    send_to_children: bool,
-) -> Result<bool> {
-    let signal = map_action_to_signal(action);
-    send_signal_to_child(child, signal, send_to_children)
-}
-
-/// Send a signal to one of Pueue's child process handles.
-///
-/// There are two scenarios:
-///
-/// **Normal case**
-///
-/// A task, such as `sleep 60` get's spawned by the posix shell `sh`.
-/// This results in the process `sh -c 'sleep 60'`.
-/// Since the posix shell doesn't propagate any process signals to its children, we have to:
-/// 1. Send the signal to the shell.
-/// 2. Send the signal directly to the children.
-///     In our case this would be the `sleep 60` child process.
-///
-/// If the user also want's to send the signal to all child processes of the task,
-/// we have to get all child-processes of the child process.
-///
-/// **Special case**
-///
-/// The posix shell `sh` has some some inconsistent behavior.
-/// In some circumstances and environments, the `sh -c $command` doesn't spawn a `sh` process with a
-/// `$command` child-process, but rather spawns the `$command` as a top-level process directly.
-///
-/// This makes things a bit more complicated, since we have to find out whether a shell is spawned
-/// or not. If a shell is spawned, we do the **Normal case** handling.
-///
-/// If **no** shell is spawned, we have to send the signal to the top-level process only.
-///
-/// If the user also want's to send the signal to all child processes of the task,
-/// we have to get all child-processes of that `$command` process. and send them the signal.
-///
-/// Returns `Ok(true)`, if everything went alright
-/// Returns `Ok(false)`, if the process went away while we tried to send the signal.
```

Comment:
You removed all of this logic, but this problem sadly doesn't go away.

For example when executing multiple commands after another `sleep 1 && sleep 1`, a root `sh -c` command will be spawned, which the `sleep 1` processes attach to.

For single stand-alone processes however, `sh -c` directly replaces itself with the target command.
Shells don't forward signals by default, which means that signals won't be propagated as expected.

For the purpose of convenience, I did all of this logic to spare the user from having to add a `-c` flag depending on whether they run a something that will result in a `sh -c` process or not.

I'm very much in favor of re-adding this behavior.